src/main/java/com/marklogic/appdeployer/command/ResourceManagerImpl.java

```diff
@@ -0,0 +1,161 @@
+package com.marklogic.appdeployer.command;
+
+import com.marklogic.client.ext.helper.LoggingObject;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Properties;
+
+/**
+ * Defines operations for managing whether a module needs to be installed or not.
+ */
+public class ResourceManagerImpl extends LoggingObject implements ResourceManager {
+
+	public static final String DEFAULT_FILE_PATH = "build/ml-gradle/resource-timestamps.properties";
+
+	private static MessageDigest md5Digest;
+	static {
+		try {
+			md5Digest = MessageDigest.getInstance("MD5");
+		} catch (NoSuchAlgorithmException e) {
+			e.printStackTrace();
+		}
+	}
+
+	private Properties props;
+	private String hashFilePath;
+
+	public ResourceManagerImpl() {
+		this(DEFAULT_FILE_PATH);
+	}
+
+	public ResourceManagerImpl(String propertiesFilePath) {
+		props = new Properties();
+		this.hashFilePath = propertiesFilePath;
+		initialize();
+	}
+
+	@Override
+	public void initialize() {
+		File propertiesFile = new File(hashFilePath);
+		logger.info("Loading properties from: " + propertiesFile.getAbsolutePath());
+		propertiesFile.getParentFile().mkdirs();
+		if (propertiesFile.exists()) {
+			FileInputStream fis = null;
+			try {
+				fis = new FileInputStream(propertiesFile);
+				if (logger.isDebugEnabled()) {
+					logger.debug("Loading properties from: " + propertiesFile.getAbsolutePath());
+				}
+				props.load(fis);
+			} catch (Exception e) {
+				logger.warn("Unable to load properties, cause: " + e.getMessage());
+			} finally {
+				try {
+					fis.close();
+				} catch (Exception e) {
+					logger.warn(e.getMessage());
+				}
+			}
+		}
+	}
+
+	@Override
+	public boolean hasFileBeenModifiedSinceLastDeployed(File file) {
+		String lastDeployedChecksum = null;
+		try {
+			lastDeployedChecksum = computeFileChecksum(md5Digest, file);
+		} catch (IOException ie) {
+			throw new RuntimeException("Unable to compute hash on file from path: " + file.getAbsolutePath() + "; cause: " + ie.getMessage(), ie);
+		}
+		if (lastDeployedChecksum != null) {
+			String key = buildKey(file);
+			String lastChecksum = props.getProperty(key);
+			if (logger.isDebugEnabled()) {
+				logger.debug("lastDeployedChecksum: " + lastDeployedChecksum);
```

Comment:
Make this one log statement, with the path of the file being included; e.g. "File: (path); key: (key); lastDeployedChecksum: (value); lastChecksum: (value)"